astropy/wcs/tests/test_utils.py

```diff
@@ -353,6 +370,59 @@ def test_wcs_to_celestial_frame():
     assert isinstance(frame, Galactic)
 
 
+def test_wcs_to_body_frame():
+    # Import astropy.coordinates here to avoid circular imports
+    from astropy.coordinates import BaseCoordinateFrame
+
+    unknown_wcs = WCS(naxis=2)
+    unknown_wcs.wcs.ctype = ["UTLN-TAN", "UTLT-TAN"]
+    with pytest.raises(KeyError, match="unknown solar system object abbreviation UT"):
+        frame = wcs_to_celestial_frame(unknown_wcs)
+
+    triaxial_wcs = WCS(naxis=2)
+    triaxial_wcs.wcs.ctype = ["MELN-TAN", "MELT-TAN"]
+    if not WCSLIB_LT8:
+        triaxial_wcs.wcs.aux.a_radius = 2439700.0
+        triaxial_wcs.wcs.aux.b_radius = 2439900.0
+        triaxial_wcs.wcs.aux.c_radius = 2438800.0
+    with pytest.raises(
+        NotImplementedError, match="triaxial systems are not supported at this time"
+    ):
+        frame = wcs_to_celestial_frame(triaxial_wcs)
+
+    mywcs = WCS(naxis=2)
+    mywcs.wcs.ctype = ["MALN-TAN", "MALT-TAN"]
+    mywcs.wcs.radesys = "ICRS"
```

Comment:
I'm still confused why it is necessary to set this.

Reply:
Removed them: it is not necessary, indeed.